### DISCUSSION TITLE
mate-session cannot find display

### DISCUSSION THREAD

**Discussion by ToddAndMargo** [mate-desktop/mate-desktop#629] [Q&A] [answered]
Fedora 39
mate-utils-common-1.26.1-3.fc39.noarch
xorg-x11-server-Xorg-1.20.14-35.fc39.x86_64

With lightdm removed from my system, when I run "mate-session" from the a user's command line, I get
```
$ mate-session
** (mate-session:8363): WARNING **: 20:26:30.590: Cannot open display:
```

How do I troubleshoot this?

**Comment by lukefromdc** [comment]
Pretty sure the session would need to be started with xinit or startx, been a while so not sure.
The session itself does not start the xserver

**Reply by ToddAndMargo** [reply]
What server should I be looking for?

Is it one of these?

```
# ps ax | grep -i [s]erver
<nothing>

# ps ax | grep -i [x]11
<nothing>

# ps ax | grep -i [x]org
  26212 ?        Ss     0:00 /usr/bin/abrt-dump-journal-xorg -fxtD
  27403 tty2     Sl     0:46 /usr/libexec/Xorg :0 vt2 vt2

```

**Reply by ToddAndMargo** [reply]
Not to ask to dumb a question, but is "mate-session" the proper way to start MATE from the command line?

**Comment by lukefromdc** [comment]
I really don't have the knowledge to answer that one, maybe someone else here does.

**Comment by lukefromdc** [comment]
By itself no, but as I said I don't have much experience doing this.

**Reply by ToddAndMargo** [reply]
I also have a virtual machine of Fedora 41 and MATE.  I see this running:

```
$ ps ax | grep -i [x]org
    715 ?        S<Ls   0:00 /usr/bin/earlyoom -r 0 -m 4 -M 409600 --prefer ^(Web Content|Isolated Web Co)$ --avoid ^(dnf|packagekitd|gnome-shell|gnome-session-c|gnome-session-b|lightdm|sddm|sddm-helper|gdm|gdm-wayland-ses|gdm-session-wor|gdm-x-session|Xorg|Xwayland|systemd|systemd-logind|dbus-daemon|dbus-broker|cinnamon|cinnamon-sessio|kwin_x11|kwin_wayland|plasmashell|ksmserver|plasma_session|startplasma-way|sway|i3|xfce4-session|mate-session|marco|lxqt-session|openbox|cryptsetup)$
    812 ?        Ss     0:00 /usr/bin/abrt-dump-journal-xorg -fxtD
   1019 tty1     Ssl+   0:01 /usr/libexec/Xorg -core -noreset :0 -seat seat0 -auth /run/lightdm/root/:0 -nolisten tcp vt1 -novtswitch

```

Does that give a hint?

**Comment by ToddAndMargo** [comment, marked as answer]
Cancel the request. It is an issue with lightdm. I replaced it with lxdm and problems solved.

I am stuck in the "endless loop" problem that is plaguing lightdm users. I have searched my ass off an everyone
with the issue can not find a cure.

So I disabled lightdm and installed lxdm. It ain't as pretty BUT IT WORKS!!!!

I even have autologin configured. And it lets me switch back and forth between Xfce and MATE.

My keeper on lxdm:

```
lxdm auto login:

# vi /etc/lxdm/lxdm.conf

[base]
## uncomment and set autologin username to enable autologin
# autologin=todd
autologin=todd


## default session or desktop used when no systemwide config
# session=/usr/bin/startlxde
session=/usr/bin/startxfce4

```
